### PR TITLE
[Xamarin.Android.Build.Tasks] Fix Ndk .exe lookup.

### PR DESCRIPTION
**Pull request by dellis1972**
The r19 Ndk on windows does not use `.exe` files
for some of its compiler tools. It uses `.cmd`.
As a result our current setup will not find these
files and you end up with errors like

	error XA5101: Toolchain utility 'armv7a-linux-androideabi16-clang.exe' for target Arm was not found. Tried in path: "C:\Users\dlab14\android-toolchain\ndk\toolchains\llvm\prebuilt\windows-x86_64"

This commit changes the code to use the
`MonoAndroidHelper.GetExecutablePath` method which
uses the `PATHEXT` environment variable to search
for various exe file extensions.

### CODE REVIEW

**Comment by jonathanpeppers** [comment]
Yeah this PR looks good, but the AOT-related tests are still failing on Windows:
image

These are happening on master, too.

So maybe there are other issues?

File: src/Xamarin.Android.Build.Tasks/Tasks/NdkUtils.cs [hunk 1]
```diff
@@ -57,6 +57,7 @@ public static string GetNdkToolPrefix (string androidNdkPath, AndroidTargetArch
 
 		public static string GetNdkTool (string androidNdkPath, AndroidTargetArch arch, string tool, int apiLevel)
 		{
+			System.Diagnostics.Debugger.Break ();
```
Comment (Therzok) [inline src/Xamarin.Android.Build.Tasks/Tasks/NdkUtils.cs]: Debug code?

**Comment by jonpryor** [comment]
I think this PR was obsoleted by PR #2664?

File: src/Xamarin.Android.Build.Tasks/Tasks/NdkUtils.cs [hunk 2]
```diff
@@ -153,7 +153,9 @@ public static string GetNdkTool (string androidNdkPath, AndroidTargetArch arch,
 				toolName = $"{toolName}{extension}";
 			else
 				toolName = $"{toolchainPrefix}-{toolName}{extension}";
-			string toolPath  = Path.Combine (toolchainDir, "bin", toolName);
+
+			string toolExe  = MonoAndroidHelper.GetExecutablePath (toolchainDir, toolName);
```
Comment (jonpryor) [inline src/Xamarin.Android.Build.Tasks/Tasks/NdkUtils.cs]: This change doesn't seem semantically equivalent: the previous version looks at `{toolchainDir}/bin/{toolName}`, while the new version looks at `{toolchainDir}/{toolName}%EXT%`.  The `bin` directory has been removed.

That said, the [Windows build succeeded](
https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_build/results?buildId=2370254)...but I don't think it actually ran any unit tests.

Let's say I'm "concerned" and would like verification of the correct path? :-)

**Comment by jonpryor** [comment]
The [`TestNdkUtil` test failed](https://jenkins.mono-project.com/job/xamarin-android-pr-builder-release/592/testReport/junit/Xamarin.Android.Build.Tests.Tasks/NdkUtilTests/TestNdkUtil___Release/):

```
  Min Api Level for X86 should be 14.
  Expected: 14
  But was:  16
```

Is the test correct?

**Pull request merged**